config.cpp

```diff
@@ -41,33 +44,44 @@ Registry::Registry()
 	);
 }
 
+Registry::~Registry() {
+	RegCloseKey(m_reg_key);
+}
```

Comment:
The `Registry` class was pretty weird. It never closed the key it created, but it also wasn't a singleton, so it was extremely unclear where and when it was safe to use. I made it close the key once destructed so it has more RAII/transactional-ish semantics.